keps/sig-node/NNNN-memory-manager-hugepages-verification/README.md

```diff
@@ -0,0 +1,478 @@
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes enhancing the Memory Manager's Static policy to verify OS-reported
+free hugepages availability during pod admission. Currently, the Memory Manager only
+tracks hugepage allocations for Guaranteed QoS pods but doesn't verify actual
+hugepage availability from the operating system. This can lead to pods being admitted
+when hugepages aren't actually available, causing runtime failures.
+
+The enhancement adds verification by reading free hugepages from sysfs
+(`/sys/devices/system/node/node<N>/hugepages/hugepages-<size>kB/free_hugepages`)
+during pod admission, ensuring pods requesting hugepages are only admitted when
+sufficient free hugepages exist.
+
+## Motivation
+
+The Memory Manager tracks hugepage allocations for Guaranteed QoS pods to provide
+NUMA-aware memory and hugepage pinning. However, it operates on its internal
+accounting without verifying the actual state of hugepages on the system.
+
+This creates a problem when:
+1. Burstable or BestEffort pods consume hugepages (via hugetlbfs mounts or
+   `mmap` with `MAP_HUGETLB`) without being tracked by the Memory Manager
+2. External processes or other system components consume hugepages
+3. The Memory Manager's internal state becomes stale or inconsistent with reality
+
+In these scenarios, a Guaranteed pod requesting hugepages may be admitted based
+on the Memory Manager's internal tracking, only to fail at runtime when the
+container attempts to use the already-exhausted hugepages.
+
+### Goals
+
+- Verify OS-reported free hugepages during pod admission for the Static policy
+- Reject pods requesting hugepages when insufficient free hugepages are available
+- Provide clear error messages when admission fails due to insufficient hugepages
+- Maintain backwards compatibility with existing Memory Manager behavior
+
+### Non-Goals
+
+- Track hugepage usage by Burstable or BestEffort pods in the Memory Manager
+- Modify scheduler behavior or add hugepage awareness to the scheduler
+- Provide hugepage reservation or preemption mechanisms
+- Support platforms other than Linux
```

Comment:
We will likely need some conversation about this and sync with sig-node/sig-windows